Packs/DeepInstinct/Integrations/DeepInstinct/DeepInstinct.py

```diff
@@ -0,0 +1,324 @@
+    file_hash = demisto.args().get('file_hash')
+    http_request('DELETE', '/policies/%s/blacklist/hashes/%s' % (str(policy_id), file_hash))
+    demisto.results('ok')
+
+
+def remove_hash_from_whitelist():
+    """
+    Remove hash from whitelist
+    """
+    policy_id = demisto.args().get('policy_id')
+    file_hash = demisto.args().get('file_hash')
+    http_request('DELETE', '/policies/%s/whitelist/hashes/%s' % (str(policy_id), file_hash))
+    demisto.results('ok')
+
+
+def add_devices_to_group():
+    """
+    Add devices to specific group
+    """
+    group_id = demisto.args().get('group_id')
+    device_ids_input = demisto.args().get('device_ids')
+    device_ids = [int(num) for num in device_ids_input.split(",")]
+    http_request('POST', '/groups/%s/add-devices' % str(group_id), json={"devices": device_ids})
+    demisto.results('ok')
+
+
+def remove_devices_from_group():
+    """
+    Remove devices from group
+    """
+    group_id = demisto.args().get('group_id')
+    device_ids_input = demisto.args().get('device_ids')
+    device_ids = [int(num) for num in device_ids_input.split(",")]
+
+    http_request('POST', '/groups/%s/remove-devices' % str(group_id), json={"devices": device_ids})
+    demisto.results('ok')
+
+
+def delete_files_remotely():
+    """
+    Delete given file ids remotely
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/devices/actions/delete-remote-files', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def terminate_remote_processes():
+    """
+    Terminate remove processes by given event ids
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/devices/actions/terminate-remote-process', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def close_events():
+    """
+    Close events by event ids
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/events/actions/close', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def fetch_incidents():
+    incidents = []
+    last_id = demisto.params().get('first_fetch_id')
+
+    last_run = demisto.getLastRun()
+    if last_run and last_run.get('last_id') is not None:
+        last_id = last_run.get('last_id')
+
+    events = http_request('GET', '/events/?after_id=' + str(last_id))
+    while events and events['events']:
+        for event in events['events']:
+            incident = {
+                'name': "DeepInstinct_" + str(event['id']),  # name is required field, must be set
+                'occurred': event['insertion_timestamp'],
+                'rawJSON': json.dumps(event)
+            }
+            incidents.append(incident)
+
+        demisto.setLastRun({'last_id': events['last_id']})
```

Comment:
need to make sure that the id you are setting is bigger than the previous one

Reply:
why? how can it be otherwise?

Reply:
Are your IDs always a number in a growing order? If so dismiss this one